libraries/ktx/src/ktx/Reader.cpp

```
@@ -0,0 +1,186 @@
+            src += keyValueByteSize + keyValuePadding;
+        }
+
+        return keyValues;
+    }
+
+    Images getImagesTable(const Header& header, size_t mipsDataSize, const Byte* mipsData) {
+        Images images;
+        auto currentPtr = mipsData;
+        auto numMips = header.getNumberOfLevels();
+
+        // Keep identifying new mip as long as we can at list query the next imageSize
+        while ((currentPtr - mipsData) + sizeof(uint32_t) <= (mipsDataSize)) {
+
+            // Grab the imageSize coming up
+            size_t imageSize = *reinterpret_cast<const uint32_t*>(currentPtr);
+            currentPtr += sizeof(uint32_t);
+
+            // If enough data ahead then capture the pointer
+            if ((currentPtr - mipsData) + imageSize <= (mipsDataSize)) {
+                auto padding = Header::evalPadding(imageSize);
+
+                images.emplace_back(Image(imageSize, padding, currentPtr));
+
+                currentPtr += imageSize + padding;
+            } else {
+                break;
+            }
+        }
+
+        return images;
+    }
+
+
+    bool KTX::checkHeaderFromStorage(const Storage& src) {
+        try {
+            auto srcSize = src.size();
+            auto srcBytes = src.data();
+
+            // validation
+            if (srcSize < sizeof(Header)) {
+                throw ReaderException("length is too short for header");
+            }
+            const Header* header = reinterpret_cast<const Header*>(srcBytes);
+
+            if (!checkIdentifier(header->identifier)) {
+                throw ReaderException("identifier field invalid");
+            }
+
+            bool endianMatch { true };
+            if (!checkEndianness(header->endianness, endianMatch)) {
+                throw ReaderException("endianness field has invalid value");
+            }
+
+            // TODO: endian conversion if !endianMatch - for now, this is for local use and is unnecessary
+
```

for now, if !endianMatch, we should throw an exception